index.js

```diff
@@ -48,11 +50,52 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
-  clangFormatProcess.on('close', function(exit) {
-    if (exit) done(exit);
-  });
-  return clangFormatProcess;
+
+  // extract glob, if present
+  var filesGlob = args.filter(function(arg){return arg.indexOf('--glob=') !== -1;})
+                    .map(function(arg){return arg.replace('--glob=', '');})
+                    .shift();
+
+  if (filesGlob) {
+    // remove glob from arg list
+    args = args.filter(function(arg){return arg.indexOf('--glob=') === -1;});
+
+    return glob(filesGlob, function(er, files) {
+      // split file array into chunks of 30
+      var i,j, chunks = [], chunkSize = 30;
+      for (i=0,j=files.length; i<j; i+=chunkSize) {
+          chunks.push( files.slice(i,i+chunkSize));
+      }
+
+      // launch a new process for each chunk
+      async.series(
+        chunks.map(function(chunk) {
+          return function(callback) {
+            var clangFormatProcess = spawn(nativeBinary,
+                                          args.concat(chunk),
+                                          {stdio: stdio});
+            clangFormatProcess.on('close', function(exit) {
+              if (exit !== 0) callback(exit);
+              else callback(null, exit);
+            });
+          };
+        }),
+        function(err, results) {
+          if (err) done(err);
+          console.log('\n');
+          console.log('ran clang-format with',
```

Comment:
Is the message appearing as duplicate for you? It should only output once, as it does on my setup. The newline is there because otherwise it would appear without separation from the formatted content (when not in `-i` mode).

As far as I can tell, the only output from `clang-format` proper is an exit code, so the individual formats (or lack thereof) aren't accessible.